manticore/core/state.py

```diff
@@ -231,6 +231,19 @@ def solve_n(self, expr, nsolves, policy='minmax'):
         '''
         return self._solver.get_all_values(self.constraints, expr, nsolves, silent=True)
 
+    def solve_buffer(self, addr, nbytes):
+        '''
+        Reads `nbytes` of symbolic data from a buffer in memory at `addr` and attempts to
+        concretize it
+
+        :param int address: Address of buffer to conretize
+        :param int nbytes: Size of buffer to conretize
+        :return: Concrete contents of buffer
+        :rtype: list[int]
+        '''
+        buffer = self.cpu.read_bytes(addr, nbytes)
+        return ''.join(chr(self.solve_one(x)) for x in buffer)
```

Comment:
I see. Even though this will normally work, I'd say we should still strive to make it as sound as possible. Basically as I understand it, after solving each character, to we need to add a constraint back on the state for the solution, so the solver can take that into account, if there happen to be relationships among the bytes in the buffer.

We could implement this right here in `solve_buffer`, but perhaps also modify a deeper interface like solver.get_value as felipe suggested.